crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -38,18 +36,24 @@ pub(crate) enum Instruction {
 
     /// Performs a function call with a list of its arguments.
     Call { func: FunctionId, arguments: Vec<ValueId> },
+
     /// Performs a call to an intrinsic function and stores the
     /// results in `return_arguments`.
     Intrinsic { func: IntrinsicOpcodes, arguments: Vec<ValueId> },
 
+    /// Allocates a region of memory. Note that this is not concerned with
+    /// the type of memory, the type of element is determined when loading this memory.
+    ///
+    /// `size` is the size of the region to be allocated by the number of FieldElements it
+    /// contains. Note that non-numeric types like Functions and References are counted as 1 field
+    /// each.
+    Allocate { size: u32 },
```

Comment:
To be clear, this will allocate `size` number of field elements -- (We might want to make this clear as when I read "Allocates a region of memory" I immediately thought of bytes)

Reply:
Yes, it is the number of FieldElements. I thought the comment explained it in the second paragraph:

> `size` is the size of the region to be allocated by the number of FieldElements it contains

Do you think this should be made more clear?

Reply:
I thought it was a bit too late in the documentation but I'm just being nit-picky, so we can leave it!